code/modules/pool/pool_main.dm

```diff
@@ -0,0 +1,278 @@
+/turf/open/pool
+	icon = 'icons/turf/pool.dmi'
+	name = "poolwater"
+	desc = "You're safer here than in the deep."
+	icon_state = "pool_tile"
+	heat_capacity = INFINITY
+	var/filled = TRUE
+	var/next_splash = 1
+	var/obj/effect/overlay/water/watereffect
+	var/obj/effect/overlay/water/top/watertop
+	var/obj/machinery/pool/controller/controller
+
+
+/turf/open/pool/Initialize()
+	watereffect = new /obj/effect/overlay/water(src)
+	watertop = new /obj/effect/overlay/water/top(src)
+	. = ..()
+
+/turf/open/pool/Destroy()
+	QDEL_NULL(watereffect)
+	QDEL_NULL(watertop)
+	controller = null
+	return ..()
+
+/turf/open/pool/proc/update_icon()
+	if(!filled)
+		name = "drained pool"
+		desc = "No diving!"
+		QDEL_NULL(watereffect)
+		QDEL_NULL(watertop)
+	else
+		name = "poolwater"
+		desc = "You're safer here than in the deep."
+		watereffect = new /obj/effect/overlay/water(src)
+		watertop = new /obj/effect/overlay/water/top(src)
+
+/obj/effect/overlay/water
+	name = "water"
+	icon = 'icons/turf/pool.dmi'
+	icon_state = "bottom"
+	density = 0
+	mouse_opacity = 0
+	layer = ABOVE_MOB_LAYER
+	anchored = TRUE
+
+/obj/effect/overlay/water/top
+	icon_state = "top"
+	layer = BELOW_MOB_LAYER
+
+/mob/living
+	var/swimming = FALSE
+
+//Put people out of the water
+/turf/open/floor/MouseDrop_T(mob/living/M, mob/living/user)
+	if(user.stat || user.lying || !Adjacent(user) || !M.Adjacent(user)|| !iscarbon(M))
+		if(issilicon(M))
+			var/turf/T = get_turf(M)
+			if(istype(T, /turf/open/pool))
+				M.visible_message("<span class='notice'>[M] begins to float.", \
+					"<span class='notice'>You start your emergency floaters.</span>")
+				if(do_mob(user, M, 20))
+					M.forceMove(src)
+					to_chat(user, "<span class='notice'>You get out of the pool.</span>")
+		return ..()
+	if(!M.swimming) //can't put yourself up if you are not swimming
+		return ..()
+	if(user == M)
+		M.visible_message("<span class='notice'>[user] is getting out the pool", \
+						"<span class='notice'>You start getting out of the pool.</span>")
+		if(do_mob(user, M, 20))
+			M.swimming = FALSE
+			M.forceMove(src)
+			to_chat(user, "<span class='notice'>You get out of the pool.</span>")
+	else
+		user.visible_message("<span class='notice'>[M] is being pulled to the poolborder by [user].</span>", \
+						"<span class='notice'>You start getting [M] out of the pool.")
+		if(do_mob(user, M, 20))
+			M.swimming = FALSE
+			M.forceMove(src)
+			to_chat(user, "<span class='notice'>You get [M] out of the pool.</span>")
+			return
+
+/turf/open/floor/CanPass(atom/movable/A, turf/T)
```

Comment:
what is all this extra canpass crap? why are you adding more stuff to the floor path and not the pool path?

Reply:
I know why this was done, but it's still bad code. I don't really know how to fix it though.
It was done for handling of movement in and out of the pool. You can't just walk out of it, you need to click and drag yourself out or use a ladder.